Packit: add jobs for downstream Fedora package builds

lsm5 opened the pull request:
Fixes: #231

lachmanfrantisek commented on .packit.yaml:
```diff
@@ -69,3 +69,22 @@ jobs:
+    trigger: commit
+    dist_git_branches:
+      - fedora-all
+    allowed_pr_authors:
+      - packit
+    allowed_committers:
+      - packit
```
Packit is allowed by default. So, you don't need those lines.
```suggestion
```

lachmanfrantisek reviewed (approved):
:+1:

lsm5 commented:
@rhatdan let's get this in before you cut the next release

lsm5 commented:
> +1

@lachmanfrantisek thanks for the review. I added your change too.

lsm5 commented:
hmm, the packit docs say by default it will modify the release in spec. Gotta set that to false.

lachmanfrantisek commented on .packit.yaml:
```diff
@@ -69,3 +69,19 @@ jobs:
+
+  - job: propose_downstream
+    trigger: release
+    update_release: false
```
The update release option does not work with `propose_downstream`, but since you use `autorelease` macro downstream, this should be preserved so I think the default behaviour should be fine.

If you want to see the result, you can use [ `packit propose-downstream --local-content` via CLI](https://packit.dev/docs/cli/propose-downstream/) to open the PRs so you don't need to wait for the new release.

lsm5 replied:
@lachmanfrantisek ack, i think downstream tasks are hard to do if  `container-selinux.spec` is not in the upstream checkout, correct? Currently, we have `container-selinux.spec.rpkg` upstream along with a:
```
post-upstream-clone:
    - rpkg spec --outdir ./
```

I think it might be easiest for us to just switch from `.spec.rpkg` to `.spec` upstream and have it sync with downstream via packit.

lsm5 commented:
moved the spec file to `rpm/` dir and set Version in it to 0 by default. Also added a helper script there for anyone looking to do local rpm builds with the latest tagged version.

The current version of the PR also allows setting Epoch differently on our copr (for copr_username: `rhcontainerbot`) and on koji. So, new rpm builds on copr and koji will not break existing users.

lsm5 commented:
@lachmanfrantisek I made more changes to this PR to be in line with standard packit workflow. There's no more `container-selinux.spec.rpkg` now. The `rpm/container-selinux.spec` will now be synced with the `container-selinux.spec` in the official Fedora package  once a new version is cut. Could you give it 1 final look please?

lsm5 commented:
@rhatdan PTAL. I guess this is good to go. The spec file in `rpm/` subdir will be synced with the official fedora package via a PR like: https://src.fedoraproject.org/rpms/container-selinux/pull-request/10 that would need a human to review and merge. That's the only human step apart from monitoring bodhi updates.

 If Frantisek has any further comments, I can work on those in a followup PR.

rhatdan commented:
LGTM
